operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -200,6 +262,75 @@ private String getLockName(String namespace, String name) {
         return result.future();
     }
 
+
+
+    /**
+     * Updates the Status field of the Kafka CR. It diffs the desired status against the current status and calls
+     * the update only when there is any difference in non-timestamp fields.
+     *
+     * @param desiredStatus The KafkaStatus which should be set
+     *
+     * @return
+     */
+    Future<Void> updateStatus(Reconciliation reconciliation, S desiredStatus, List<Condition> unknownAndDeprecatedConditions) {
```

Comment:
I find this method signature a little odd. You're only passing `unknownAndDeprecatedConditions` so you can add it to `desiredStatus`. I think it would be better to factor that into its own method which the caller of this method invokes and passes just the `desiredStatus`.